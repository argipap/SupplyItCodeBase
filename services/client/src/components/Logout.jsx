import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Logout extends Component {
    componentDidMount() {
        this.props.logoutUser();
    };
    render() {
        return (
            <div>
                <p>You are now logged out. Click <Link to="/login">here</Link> to log back in.</p>
            </div>
        )
    };
};

export default Logout;