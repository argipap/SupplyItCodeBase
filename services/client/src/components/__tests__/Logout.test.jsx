import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Logout from '../Logout';

const logoutUser = jest.fn();

test('Logout renders properly', () => {
    const wrapper = shallow(<Logout logoutUser={logoutUser}/>);
    const element = wrapper.find('p');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children[0]).toContain('You are now logged out.');
});

test('Logout renders a snapshot properly', () => {
    const tree = renderer.create(
        <Router><Logout logoutUser={logoutUser}/></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});