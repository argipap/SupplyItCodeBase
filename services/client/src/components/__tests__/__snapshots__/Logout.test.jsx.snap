// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders a snapshot properly 1`] = `
<div>
  <p>
    You are now logged out. Click 
    <a
      href="/login"
      onClick={[Function]}
    >
      here
    </a>
     to log back in.
  </p>
</div>
`;
